Comments Redesign: Add success global notices

Copons opened the pull request:
Add the success notices with the Undo feature for every action to the new `Comment` component.

- Approve, Unapprove, Spam, Trash
- Like and Approve (from unapproved)
- Reply, Reply and Approve (no Undo)
- Delete (no Undo) - not part of the PR, but it has the notice anyway.

Big changes from the current implementation:

- Notices don't stack anymore. We always have 1 single comment notice visible, referring to the last action performed.
- Notices don't disappear automatically after 5 seconds anymore, but have to be dismissed manually, because otherwise the first timeout time would be applied to all other notices 
I think this is caused because of how the notices timeout is implemented: 

### Notices Timeout

Notices duration is an odd beast.
As it happens, the first notice's timeout will be applied to all the following notices that appear in the same position in the timeout duration.
This, I think happens because notices are stored in an array.
So, let's say we create the first notice, with a duration of 5s.
After 3 seconds we dispatch a new notice; the first notice timeout has 2s left.
We remove the first notice because it's not relevant anymore, and add the new notice.
The second notice is added to the notices array at the same index of the first notice, and for some reasons will be unmounted when the first notice's timeout expires.
This means, that the second notice will only last 2s instead of 5s; in other words, it will expire at T+5s instead of T+8s as expected.

### Last Undo

You might notice a `lastUndo` logic brought all the way from `CommentList`.
This is a small workaround required to avoid a race condition happening when undoing persisted comments, causing the comment to disappear when is removed from the persisted list.

### Testing instructions

Checkout this branch and run it with
`ENABLE_FEATURES=comments/management/m3-design npm start`

rodrigoi commented:
> The second notice is added to the notices array at the same index of the first notice, and for some reasons will be unmounted when the first notice's timeout expires.
Maybe you can create an issue, but I have the feeling that somehow, that's by design...

gwwar commented:
> This is a small workaround required to avoid a race condition happening when undoing persisted comments, causing the comment to disappear when is removed from the persisted list.

Did we never come to a resolution around removing persistence from approved? I think you made a good point that the All filter already handles this for us.

gwwar commented on client/my-sites/comments/comment/comment-actions.jsx:
```diff
@@ -93,6 +101,64 @@ export class CommentActions extends Component {
+			button: translate( 'Undo' ),
+			id: 'comment-notice',
+			isPersistent: true,
+			onClick: () => {
```
Is it possible pulling this up, or did we need the closure here? Looks like most of this should be available from props.

Copons commented:
> I think you made a good point that the All filter already handles this for us.

@gwwar Did I? 😅 
I did try to make that point, but I can't remember how it was received. Though, I'm definitely happy to kill the persistence once and for all!

> In Mobile if the notice is visible, we aren't able to interact with the last comment.

That's the expected notices behaviour though, and it happens in the old design as well.

<img width="386" alt="screen shot 2017-11-10 at 10 25 39" src="https://user-images.githubusercontent.com/2070010/32654109-fb7f313e-c601-11e7-9c02-533bf99c1c9f.png">

Not much we can do, nor I think we should actually.

Copons commented:
Updates:
- Moved the undo into its own class method. It looks like it's working! Great suggestion @gwwar 
- Removed the persistence logic because I made a point I forgot about! 😄 

One thing about the persistence: I mentioned that `lastUndo` was a workaround for the persistence, but I actually remembered it wrong.
It's a workaround on a race condition, yes, but on how the data layer and the API response update the state. #16838 contains all the info needed.

gwwar commented:
Nice! Still tests well. Feel free to :shipit: when ready.
